### PR TITLE
tests: add postgres integration tests

### PR DESCRIPTION
**Pull request by jimlambrt**
PR #7 includes [WithPgPlaceholder() option](https://github.com/hashicorp/mql/commit/fa247ec3f9cba63a795ec08de158fd4c01d07cee)(https://github.com/hashicorp/mql/commit/fa247ec3f9cba63a795ec08de158fd4c01d07cee) and we will wait for it to be merged, before updating the mql dependency in this PR and merging.

### CODE REVIEW

File: .github/workflows/tests-postgres.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  postgres:
+    strategy:
+      matrix:
+        dbversion: ["postgres:latest"]
+        go: ["1.20", "1.19", "1.18"]
```
Comment (johanbrandhorst) [inline .github/workflows/tests-postgres.yml:14]: This can probably do with updating
```suggestion
        go: ["1.21", "1.20", "1.19"]
```

Comment (jimlambrt) [reply .github/workflows/tests-postgres.yml:14]: ty!

**Comment by jimlambrt** [comment]
I've merged #7 so we can merge this once it's reviewed/approved.

**Review by johanbrandhorst — APPROVED** [review]

File: tests/postgres/testing.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	testCtx := context.Background()
+	require.NoError(t, rw.Create(testCtx, u))
+}
+
+func testDeleteUsers(t *testing.T, rw dbw.Writer) {
+	t.Helper()
+	testCtx := context.Background()
+	_, err := rw.Exec(testCtx, "delete from users", nil)
+	require.NoError(t, err)
+}
```
Comment (mgaffney) [inline tests/postgres/testing.go:36]: There are no tests for this

Comment (jimlambrt) [reply tests/postgres/testing.go:36]: It's used in the tests which would fail if it wasn't working.

Comment (mgaffney) [reply tests/postgres/testing.go:36]: I don't see anything that calls it

Comment (mgaffney) [reply tests/postgres/testing.go:36]: And the coverage report doesn't show it being called

Comment (jimlambrt) [reply tests/postgres/testing.go:36]: I'll remove it. Ty!

**Pull request merged**